src/resolvers/claims/status-card/statusCard.ts

```diff
@@ -0,0 +1,248 @@
+import { ResolverSubset } from '../../ResolverSubset'
+import {
+  ClaimStatusCard,
+  ClaimStatusCardPill,
+  ClaimStatusCardPillType,
+  ClaimStatusProgressSegment,
+  ClaimStatusProgressType,
+} from '../../../generated/graphql'
+import {
+  ClaimDto,
+  ClaimOutcomeDto,
+  ClaimStatusDto,
+} from '../../../api/upstreams/claimService'
+import { LocalizedStrings } from '../../../translations/LocalizedStrings'
+
+export const ClaimsStatusCardsSubset: ResolverSubset = {
+  Query: {
+    claims_statusCards: async (
+      _parent,
+      args: { locale: string },
+      { upstream, strings },
+    ): Promise<ClaimStatusCard[]> => {
+      const claims = (await upstream.claimService.getMemberClaims()).sort(
+        (a, b) =>
+          new Date(a.registrationDate).getTime() -
+          new Date(b.registrationDate).getTime(),
+      )
+      const localizedStrings = strings.withLocale(args.locale)
+
+      return Promise.all(
+        claims.map((c) =>
+          transformClaimStatusCard(c, args.locale, localizedStrings),
+        ),
+      )
+    },
+  },
+}
+
+// TODO This doesn't even need to be async right? Any reason we'd keep it async if it doesn't await for anything?
+const transformClaimStatusCard = async (
+  claim: ClaimDto,
+  locale: string,
+  strings: LocalizedStrings,
+): Promise<ClaimStatusCard> => {
+  return {
+    id: claim.id,
+    pills: transformClaimStatusPills(claim, locale, strings),
+    title: transformClaimTitle(claim, strings),
+    subtitle: transformClaimSubTitle(claim),
+    progressSegments: transformProgressSegments(claim, strings),
+  } as ClaimStatusCard
```

Comment:
```suggestion
  }
```

Is this cast really necessary? I fear that we might be 'tricking' TypeScript into giving us worse type checks by making the cast.

Reply:
I thought this was like *the* way to create an instance of the `(export) type ClaimStatusCard` because none of all the other syntax I tried, like: `new ClaimStatusCard()` worked 😂
I guess you just wrap it in {} and let the IDE check it's type by comparing it with the function's return type. Interesting for sure 👀

Reply:
Yes! The concept is called _structural typing_ (as opposed to _nominal typing_). Quite neat! More reading here: https://www.typescriptlang.org/docs/handbook/type-compatibility.html